### ISSUE TITLE
`sh: yarn: not found` when installing dependencies in sub directories 

### ISSUE THREAD

**Issue by robdodson** [vercel/vercel#452]
I've split my server and client code into two different subdirectories, each with their own `package.json`. This is so I can use `yarn`'s [flat configuration option](https://yarnpkg.com/en/docs/package-json#toc-flat) in my client code.

```
my-project/
├── client
│   └── package.json
├── server
    └── package.json
package.json
```

My top level `package.json` has the following scripts:

```
"scripts": {
    "install:all": "yarn run install:client && yarn run install:server",
    "install:client": "cd ./client && yarn",
    "install:server": "cd ./server && yarn",
    "build": "yarn run install:all && tsc -p ./client/src/",
    "start": "node server/bin/www"
  }
```

Unfortunately when it gets to the `build` script it throws saying it can't find `yarn`.

```
> ▲ npm run build
> > my-project@1.0.0 build /home/nowuser/src
> > yarn run install:all && tsc -p ./client/src/
> sh: yarn: not found
```

I saw in https://github.com/zeit/now-cli/issues/153#issuecomment-279761171 that `yarn` isn't actually being used so I'm not sure if it's possible to work around this.

**Comment by TooTallNate** [comment]
Correct that `yarn` is not installed in the Node.js deployment environment.

One option could be to run `npm install yarn` in your build script and then it would be available in your npm scripts. Another option might be to set up a Dockerfile with yarn installed.

**Comment by jamo** [comment]
Or just use `npm` for invoking the commands, you may define `now-build` which will be used instead of `build` and similarly `now-start` that will be used instead of `start` script if present if you need to run different commands in now than in local setup.

**Comment by jamonholmgren** [comment]
Any chance yarn might be included in the deployment environment by default? It's a fairly common use case these days.

**Comment by ajhyndman** [comment]
I noticed that if you include a `yarn.lock` file, now deployments will introduce yarn to the dev environment to respect that.

However, I recently migrated my project to use yarn workspaces.  This means that the `yarn.lock` file no longer sits in the directory of the package I wish to deploy.  My build script refers to `yarn` and was running without issue, but now fails.

This implicit dependency injection based on the presence of a `yarn.lock` file feels a little weird.  I would also be in favour of just including `yarn` by default at this point.
